src/figuresClasses.ts

```diff
@@ -1,11 +1,98 @@
-export interface Figure {}
+export interface Figure {
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: string;
+
+  color: string;
+
+  private a: number;
+
+  private b: number;
+
+  private c: number;
+
+  constructor(color: string, a: number, b: number, c: number) {
+    this.shape = 'triangle';
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid measurements');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid measurements');
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const s = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(s * (s - this.a) * (s - this.b) * (s - this.c));
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  shape: string;
+
+  color: string;
 
-export class Triangle implements Figure {}
+  private radius: number;
 
-export class Circle implements Figure {}
+  constructor(color: string, radius: number) {
+    this.shape = 'circle';
+    this.color = color;
 
-export class Rectangle implements Figure {}
+    if (radius <= 0) {
+      throw new Error('Invalid measurements');
+    }
+
+    this.radius = radius;
+  }
+
+  getArea(): number {
+    const area = Math.PI * this.radius * this.radius;
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Rectangle implements Figure {
+  shape: string;
+
+  color: string;
```

Comment:
[CODE STYLE] - The `shape` property in the `Rectangle` class should have a literal type 'rectangle' instead of a generic string, as the shape is not meant to change. Consider using the literal type to enforce this constraint.